src/main/kotlin/gg/essential/vigilance/gui/settings/DropDownComponent.kt

```diff
@@ -0,0 +1,203 @@
+package gg.essential.vigilance.gui.settings
+
+import gg.essential.elementa.components.*
+import gg.essential.elementa.constraints.*
+import gg.essential.elementa.constraints.animation.Animations
+import gg.essential.elementa.dsl.*
+import gg.essential.elementa.effects.ScissorEffect
+import gg.essential.elementa.state.BasicState
+import gg.essential.elementa.state.State
+import gg.essential.elementa.state.toConstraint
+import gg.essential.universal.USound
+import gg.essential.vigilance.gui.VigilancePalette
+import gg.essential.vigilance.gui.common.shadow.ShadowIcon
+import gg.essential.vigilance.utils.*
+import gg.essential.vigilance.utils.ReadyOnlyState
+import gg.essential.vigilance.utils.and
+import gg.essential.vigilance.utils.hoveredState
+
+class DropDownComponent(
+    initialSelection: Int,
+    private val options: List<String>,
+    private val maxDisplayOptions: Int = 6,
+) : UIBlock() {
+
+    private val writableExpandedState: State<Boolean> = BasicState(false)
+    private val optionContainerHeight = 18
```

Comment:
This is different from what we are using in Essential (20). Why is that? If it's intentional, then should we maybe allow it to be configured via a constructor parameter?